### DISCUSSION TITLE
Тестовый стенд Flexberry.AsyncOpenXmlReports.Sample

### DISCUSSION THREAD

**Discussion by AntonIV8** [Flexberry/rfcs#24] [General]
Целью итерации является создание стенда с тестовым приложением, которое будет поддерживать следующий функционал

1) Формирование отчетов с использованием пакета NewPlatform.Flexberry.OpenXmlPowerTools.Reports
2) Загрузка/выгрузка шаблонов для отчетов
3) Формирование отчетов в асинхронном режиме через QuartzNet (с использованием технологических сервисов внутри джобов)
4) Оповещение пользователя о готовности отчета через сообщение в приложении (SIgnalR)
5) Оповещение пользователя по email (с приложением файла отчета, если небольшой)
6) Система авторизации KeyCloak

Общая схема процесса запроса и получения:
![image](https://user-images.githubusercontent.com/13151962/230287258-8d302a14-5432-4469-bf71-341356ea55fc.png)

**Comment by mao29** [comment]
1. Предлагаю добавить порядок вызовов на стрелки
2. Предлагаю заложить какую-то модель для хранения состояния джобов, запущенных асинхронно (Flexberry Jobs??). Тогда пользователь во-первых сможет на фронте видеть список и статусы запущенных джобов, во-вторых в будущем возвращаться к ранее запускавшимся джобам и скачивать результаты повторно, в-третьих, в уведомлении по вебсокету и в письме можно будет присылать ссылку на страницу джоба для просмотра его результатов, а не просто сам файл
3. Не очень понимаю почему `Авторизация KeyCloak` нарисована как часть фронтенда. Разве там не открывается для авторизации форма логина самого KeyCloak-а? 
4. Для оповещения по Email вроде тоже предполагалось использование недефолтного SmtpClient, а то про него уже [в документации ](https://learn.microsoft.com/en-us/dotnet/api/system.net.mail.smtpclient?view=net-7.0#remarks) написано что не рекомендуется использовать
5. Предлагаю в джобе формирования отчета добавить работу с какими-то объектами, требующими проверки полномочий. Таким образом встанет задача как-то нетривиально настроить контейнер в сервисе Quartz, чтобы для каждого джоба использовался свой набор инстансов IDataService, ISecurityManager, IUser и т.д. и при этом IUser должен как-то возвращать логин именно того пользователя, который инициировал запуск джоба
